backend/middleware.py

```diff
@@ -1,31 +1,37 @@
-import typing as t
-
 import ssl
 from motor.motor_asyncio import AsyncIOMotorClient
-from starlette.middleware.base import BaseHTTPMiddleware
 from starlette.requests import Request
-from starlette.responses import JSONResponse, Response
+from starlette.responses import JSONResponse
+from starlette.types import ASGIApp, Scope, Receive, Send
 
 from backend.constants import DATABASE_URL, DOCS_PASSWORD, MONGO_DATABASE
 
 
-class DatabaseMiddleware(BaseHTTPMiddleware):
-    async def dispatch(self, request: Request, call_next: t.Callable) -> Response:
+class DatabaseMiddleware:
+
+    def __init__(self, app: ASGIApp) -> None:
+        self._app = app
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         client: AsyncIOMotorClient = AsyncIOMotorClient(
             DATABASE_URL,
             ssl_cert_reqs=ssl.CERT_NONE
         )
         db = client[MONGO_DATABASE]
-        request.state.db = db
-        response = await call_next(request)
-        return response
+        scope["state"].db = db
```

Comment:
What version is this implemented in? I've tried bumping up to latest (0.20.3), but this is still a KeyError. Since this is not documented I'm not sure what you're going for.

Going through the documentation PR doesn't cover this scenario either.

Reply:
Sorry for the bonehead mistake. [`Request` generates this](https://github.com/encode/starlette/blob/master/starlette/requests.py#L166-L174), I forgot it wasn't part of the ASGI spec itself. I changed this line to just use `Request`.